cobalt/watchdog/watchdog.cc

```diff
@@ -323,46 +344,11 @@ void Watchdog::UpdateViolationsMap(void* context, Client* client,
         "violationDurationMilliseconds",
         base::Value(std::to_string(violation_duration + (time_delta / 1000))));
   }
-
   static_cast<Watchdog*>(context)->pending_write_ = true;
-}
 
-std::string Watchdog::ReadViolationFile(const char* file_path) {
-  starboard::ScopedFile read_file(file_path, kSbFileOpenOnly | kSbFileRead);
-  if (read_file.IsValid()) {
-    int64_t kFileSize = read_file.GetSize();
-    std::vector<char> buffer(kFileSize + 1, 0);
-    read_file.ReadAll(buffer.data(), kFileSize);
-    return std::string(buffer.data());
-  }
-  return "";
-}
-
-void Watchdog::InitializeViolationsMap(void* context) {
-  // Loads the previous Watchdog violations file containing violations before
-  // app start, if it exists, to populate violations_map_.
-  static_cast<Watchdog*>(context)->violations_count_ = 0;
-
-  std::string watchdog_json =
-      static_cast<Watchdog*>(context)->ReadViolationFile(
-          (static_cast<Watchdog*>(context)->GetWatchdogFilePath()).c_str());
-  if (watchdog_json != "") {
-    static_cast<Watchdog*>(context)->violations_map_ =
-        base::JSONReader::Read(watchdog_json);
-  }
-
-  if (static_cast<Watchdog*>(context)->violations_map_ == nullptr) {
-    SB_LOG(INFO) << "[Watchdog] No previous violations JSON.";
-    static_cast<Watchdog*>(context)->violations_map_ =
-        std::make_unique<base::Value>(base::Value::Type::DICTIONARY);
-  } else {
-    for (const auto& it :
-         (static_cast<Watchdog*>(context)->violations_map_)->DictItems()) {
-      base::Value& violation_dict = it.second;
-      base::Value* violations = violation_dict.FindKey("violations");
-      static_cast<Watchdog*>(context)->violations_count_ +=
-          violations->GetList().size();
-    }
+  if (static_cast<Watchdog*>(context)->violations_count_ >
```

Comment:
Do we need to keep this violations_count_ and update it all the time? It seems like this is the only place that uses it. Can we just calculate violations count here once?

Reply:
That's actually a great point, done. Much cleaner.